hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImageFormatPBINode.java

```diff
@@ -279,6 +283,9 @@ void loadINodeDirectorySection(InputStream in) throws IOException {
         INodeDirectory p = dir.getInode(e.getParent()).asDirectory();
         for (long id : e.getChildrenList()) {
           INode child = dir.getInode(id);
+          if (child.isDirectory()) {
```

Comment:
We are only incrementing here if its a directory. The inode table / section contains an entry for every file and directory in the system.

The the directory section is what links them all together into the parent child relationship, so it should contain about the same number of entries as inodes.

I am not sure if it makes sense to just count the directories here, as we have already counted them in the inode section.

Why do you want to count just directories? Would it make more sense to count each entry and child entry to give an idea of the number of entries processed by each parallel section?

Reply:
Thanks @sodonnel for the comment and review.
When loading FsImage, we have recorded the number of all inodes (including INodeFile and INodeDirectory) in the log.
For example, here is the specific record information:
`2021-09-30 19:12:55,034 [15609]-INFO [main:FSImageFormatPBINode$Loader@409]-Loading xxxx INodes.`
Yes, this is good. We can know the data of the loaded inode, but this is a sum. But we can't know how many INodeFiles or how many INodeDirectory are loaded, if we can know how many INodeFiles are loaded, similarly, we can know how many INodeDirectory is loaded. This will help us find the cause of the problem when there is an exception.
Regarding the reason for dealing with INodeDirectory here. What I want to show is that in many cases, the number of files created will be more than the number of directories created. Therefore, it may use less time when calculating INodeDirectory.